components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1819,4 +1834,42 @@ private void validateScopeExistence(String scopeName) throws IdentityOAuthAdminE
                             scopeName));
         }
     }
+
+    /**
+     * Check the requirement of having a configured public certificate or JWKS in SP and throw an exception with a
+     * specific error message if the public certificate or JWKS in SP is not configured.
+     *
+     * @param isIdTokenEncryptionEnabled Is ID token encryption is enabled or not.
+     * @param isRequestObjectSignatureValidation Is request object signature validation is enabled or not.
+     * @param clientId Client ID of the service provider.
+     * @param tenantDomain Tenant domain of the service provider.
+     * @throws IdentityOAuthClientException
+     */
+    private void handlePublicCertificateConfig(String clientId, String tenantDomain, boolean isIdTokenEncryptionEnabled,
```

Comment:
Can we have a better name to reflect the function of the method?

CheckIfPublicCertConfigured or a better name :)

Reply:
Addressed with cef5ac44bbc380cba2702c7f898071cf525ec51f